app/src/test/java/com/liato/bankdroid/DataRetrieverTaskTest.java

```diff
@@ -0,0 +1,123 @@
+package com.liato.bankdroid;
+
+import com.liato.bankdroid.banking.Bank;
+
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import android.app.ProgressDialog;
+import android.content.Context;
+import android.support.annotation.NonNull;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class DataRetrieverTaskTest {
+    private static class TestableDataRetrieverTask extends DataRetrieverTask {
+        private final Bank[] banks;
+        private final ProgressDialog dialog;
+
+        @Override
+        protected void publishProgress(int i, Bank bank) {
+            // This method intentionally left blank
+        }
+
+        @Override
+        protected Bank getBankFromDb(long bankId, Context parent, boolean loadAccounts) {
+            return banks[0];
+        }
+
+        @Override
+        protected List<Bank> getBanksFromDb(Context parent, boolean loadAccounts) {
+            return Arrays.asList(banks);
+        }
+
+        @Override
+        protected void saveBank(Bank bank, Context context) {
+            // This method intentionally left blank
+        }
+
+        @Override
+        protected boolean isContentProviderEnabled() {
+            return false;
+        }
+
+        @NonNull
+        @Override
+        protected ProgressDialog getDialog() {
+            return dialog;
+        }
+
+        /**
+         * Constructor for testing a specific bank
+         */
+        public TestableDataRetrieverTask(Bank bank) {
+            super(Mockito.mock(MainActivity.class), 5);
```

Comment:
Do you mind extracting `5` into a constant with a name like `BANK_ID` to improve readability?